app/views/profiles/_appointments.html.erb

```diff
@@ -0,0 +1,40 @@
+<% @patient = Patient.find_by medwallet_id: @user.id  %>
+<% @careplan = CarePlan.where patient_id: @patient.id %>
+<div class="table-responsive-xl">
+  <table class="table">
+    <thead>
+      <tr>
+        <th scope="col">Date</th>
+        <th scope="col">Status</th>
+        <th scope="col">Provider</th>
+        <th scope="col">Treatment</th>
+        <th scope="col">Cost</th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @careplan.each do |careplan| %>
+        <tr>
+          <td id="careplan_<%=careplan.id%>_date">
+            <%= careplan.date %>
+          </td>
+          <td id="careplan_<%=careplan.id%>_status">
+            <% @status = Status.where id: careplan.status_id %>
+            <%= link_to careplan.status.name, "care_plans/" + careplan.status_id.to_s + "/edit", remote: true %>
+        </td>
+        <td id="careplan_<%=careplan.id%>_provider">
+          <% @feeschedule = FeeSchedule.find_by id: careplan.fee_schedule_id %>
+          <% @provider = Provider.find_by id: @feeschedule.provider_id %>
+          <%= @provider.name %>
+        </td>
+        <td id="careplan_<%=careplan.id%>_action">
+          <% @treatment = TreatmentAction.find_by id: @feeschedule.treatment_id %>
+          <%= @treatment.action %>
+        </td>
+        <td id="careplan_<%=careplan.id%>_cost">
```

Comment:
I'm not sure if it's important to that these `<td>`'s have a id— but if you're open to renaming the id's to `"care_plan_<%= ... %>_status"` then you can use the `dom_id` helper method to condense some of the code.

You can also use your `belongs_to` associations here instead of `find_by`. A refactor could look something like:

```suggestion
        <td id="<%= dom_id(careplan)%>_provider">
          <% @feeschedule = careplan.fee_schedule %>
          <% @provider = @feeschedule.provider %>
          <%= @provider.name %>
        </td>
        <td id="<%= dom_id(careplan)%>_action">
          <% @treatment = @feeschedule.treatment %>
          <%= @treatment.action %>
        </td>
        <td id="<%= dom_id(careplan)%>_cost">
```

_Technically_ you don't need to make these variables instance variables either— unless you want to use them outside this View partial. I think usually that's avoided though, and instead devs prefer to pass in any required variables to the partial when it's being rendered. That way it's easier to debug.

Reply:
Made the edit!